docs/targets/mongodb.md

```diff
@@ -0,0 +1,146 @@
+
+So if one were to send the following event:
+
+```cmd
+curl -v  http://localhost:8080 \
+       -X POST \
+       -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
+       -H "Ce-Specversion: 1.0" \
+       -H "Ce-Type: io.triggermesh.sample.event" \
+       -H "Ce-Source: sample" \
+       -H "Content-Type: application/json" \
+       -d '{"example":"event"}'
+```
+
+We would expect a similar entry in the MongoDB database collection:
+
+```
+{"_id":{"$oid":"62029b4e20372fe01225194d"},"example":"event"}
+```
+
+## Pre-defined Event Types
+
+Use these event types and associated payloads to interact with the MongoDB Target.
+
+### io.triggermesh.mongodb.insert
+
+Events of this type intend to post a single key:value pair to MongoDB
+
+#### Example CE posting an event of type "io.triggermesh.mongodb.insert"
+
+```cmd
+curl -v  http://localhost:8080 \
+       -X POST \
+       -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
+       -H "Ce-Specversion: 1.0" \
+       -H "Ce-Type: io.triggermesh.mongodb.insert" \
+       -H "Ce-Source: sample/source" \
+       -H "Content-Type: application/json" \
+       -d '{"database":"test","collection": "test","mapStrVal":{"test":"testdd1","test2":"test3"}}'
+```
+
+#### This type expects a JSON payload with the following properties:
+
+| Name  |  Type |  Comment |
+|---|---|---|
+| **database** | string | The name of the database.  |
+| **collection** | string | The value of the collection. |
+| **itemName** | string | This value will be used to assing a Key.  |
+| **mapStrVal** | map[string]string | This value will be used to assing a Value. |
+
+**Note** the `database` and `collection` fields are not required. If not provided, the `defaultDatabase` and `defaultCollection` spec fields will be used.
+
+### io.triggermesh.mongodb.update
+
+Events of this type intend to update any pre-existing key:value pair(s). 
+
+#### Example CE posting an event of type "io.triggermesh.mongodb.update"
+
+```cmd
+curl -v http://localhost:8080 \
+       -X POST \
+       -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
+       -H "Ce-Specversion: 1.0" \
+       -H "Ce-Type: io.triggermesh.mongodb.update" \
+       -H "Ce-Source: sample/source" \
+       -H "Content-Type: application/json" \
+       -d '{"database":"test","collection": "test","searchKey":"test","searchValue":"testdd1","updateKey":"partstore","updateValue":"UP FOR GRABS"}'
+```
+
+#### This type expects a JSON payload with the following properties:
+
+| Name  |  Type |  Comment |
+|---|---|---|
+| **database** | string | The name of the database.  |
+| **collection** | string | The value of the collection. |
+| **itemName** | string | This value will be used to assing a Key.  |
+| **searchKey** | string | . |
+| **searchValue** | string | .  |
+| **updateKey** | string | .  |
+| **updateValue** | string |. |
```

Comment:
not sure if the dots mean they are self defined by the names. Just mentioning in case they are not.